core/src/main/java/io/undertow/server/protocol/framed/AbstractFramedChannel.java

```diff
@@ -240,9 +240,7 @@ public void run() {
             });
         } catch (RejectedExecutionException e) {
             //thread is shutting down
-            while (!taskRunQueue.isEmpty()) {
-                taskRunQueue.poll().run();
```

Comment:
I think it might be best to just create a new Thread to run these tasks. Not running them at all could also result in hangs if the task is supposed to unblock a thread or release a buffer.

Reply:
Updated.

Using the thread approach it's possible for tasks submitted by a single worker thread to be executed in parallel by two separate new threads. I suppose that case was still possible in the old model as well when multiple worker threads interacted with a channel.